src/classes/Contacts.cls

```diff
@@ -491,6 +616,135 @@ public inherited sharing class Contacts extends fflib_SObjects {
         return new Contacts(contactsInHouseholdAccounts);
     }
 
+    public List<Contact> getOneToOneContactsToUpdate() {
+        List<Contact> oneToOneContactsToUpdate = new List<Contact>();
+        for (Contact contactRecord : (List<Contact>) getRecords()) {
+            Contact oldContact = oldMap.get(contactRecord.Id);
+
+            if (needsOne2OneAccountUpdate(contactRecord, oldContact)) {
+                oneToOneContactsToUpdate.add(contactRecord);
+            }
+        }
+        return oneToOneContactsToUpdate;
+    }
+
+    public Contact oldVersionOf(Contact con) {
+        return oldMap.get(con.Id);
+    }
+
+    public List<Id> getHouseholdIdsToRename() {
+        List<Id> householdIdsToRename = new List<Id>();
+        for (Contact contactRecord : (List<Contact>) getRecords()) {
+            Contact oldContact = oldVersionOf(contactRecord);
+            if (needsHHAccountRename(contactRecord, oldContact)) {
+                householdIdsToRename.add(contactRecord.AccountId);
+            }
+        }
+        return householdIdsToRename;
+    }
+
+    public void resetDeprecatedFields() {
+        for (Contact contactRecord : (List<Contact>) getRecords()) {
+            // always clear this deprecated field to avoid confusing users.
+            contactRecord.npe01__SystemAccountProcessor__c = null;
```

Comment:
At some point we may want to consider moving this into a class-level constant variable (or somewhere else) so that we can easily track/maintain a potentially growing list of deprecated fields that need to be nulled out here.

Reply:
Yes, good call.  Let's do that when we have another field to add here.